pyrogram/client/client.py

```diff
@@ -2306,3 +2309,100 @@ def download_media(self, message: types.Message, file_name: str = None):
         self.download_queue.put((media, file_name, done))
 
         done.wait()
+
+    def add_contacts(self,
+                     phone: Union[int, str] = None,
+                     first_name: str = None,
+                     last_name: str = None,
+                     input_phone_contact_list: List[InputPhoneContact] = None):
+        if (phone is None or first_name is None) and \
+                input_phone_contact_list is None:
+            log.warning("(phone and first_name) or input_phone_contact_list "
+                        "must be not None")
+            return None
+
+        if phone is not None and first_name is not None:
+            if str(phone)[0] != '+':
+                phone = '+' + str(phone)
+            input_phone_contact_list = []
+            input_phone_contact = InputPhoneContact(client_id=0,
+                                                    phone=phone,
+                                                    first_name=first_name,
+                                                    last_name=last_name or '')
+            input_phone_contact_list.append(input_phone_contact)
+
+        # make sure that we send only InputPhoneContact
+        inner_input_phone_contact_list = []
+        for contact in input_phone_contact_list:
+            if isinstance(contact, InputPhoneContact):
+                inner_input_phone_contact_list.append(contact)
+
+        imported_contacts = self.send(
+            functions.contacts.ImportContacts(inner_input_phone_contact_list))
+
+        for user in imported_contacts.users:
+            if isinstance(user, User):
+                if user.id in self.peers_by_id:
+                    continue
+
+                if user.access_hash is None:
+                    continue
+
+                input_peer = InputPeerUser(
+                    user_id=user.id,
+                    access_hash=user.access_hash
+                )
+
+                self.peers_by_id[user.id] = input_peer
+
+                if user.username is not None:
+                    self.peers_by_username[user.username] = input_peer
+
+        return imported_contacts
+
+    def delete_contacts(self, _id: int = None,
```

Comment:
I prefer to keep just one argument (the list), like the raw function does. To add/delete a single contact you simply pass a list with a single element, e.g. [123456789]

Reply:
well, it's up to you of course

Reply:
It's also about consistency, other methods like forward_messages, and delete_messages work like this